libs/ui/widgets/src/lib/language-switcher/language-switcher.component.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Component,
+  Inject,
+  InjectionToken,
+  Input,
+  OnInit,
+  Optional,
+} from '@angular/core'
+import { LANGUAGE_STORAGE_KEY } from '@geonetwork-ui/util/i18n'
+import { TranslateService } from '@ngx-translate/core'
+
+export const LANGUAGE_PLACEHOLDER = new InjectionToken<string[]>(
+  'language-placeholder'
+)
+
+const DEFAULT_LANGUAGES = ['en', 'fr']
+
+@Component({
+  selector: 'gn-ui-language-switcher',
+  templateUrl: './language-switcher.component.html',
+  styleUrls: ['./language-switcher.component.css'],
+})
+export class LanguageSwitcherComponent implements OnInit {
+  @Input() languageList = this.languagePlaceholder || DEFAULT_LANGUAGES
+
+  constructor(
+    @Optional()
+    @Inject(LANGUAGE_PLACEHOLDER)
+    private languagePlaceholder,
+    private translate: TranslateService
+  ) {}
+
+  get currentLang() {
+    return this.translate.currentLang
+  }
+
+  ngOnInit(): void {
+    this.languageList.forEach((language, index) => {
+      this.languageList[index] = {
+        label: `language.${language}`,
+        value: language,
+      }
+    })
```

Comment:
Here you are effectively mutating the `languageList` array, which can in fact be the actual language list that was injected. So you might introduce side effects in completely unrelated modules that would depend on this injection token.

You could probably avoid mutating the injected value without too much changes like so:
```ts
const languages = this.languagePlaceholder || DEFAULT_LANGUAGES;
this.languageList = languages.map((language, index) => ({
  label: `language.${language}`,
  value: language,
}))
```